Justfile

```diff
@@ -173,4 +173,28 @@ debug *args:
   FTL_DEBUG=true just build ftl
   dlv --listen=:2345 --headless=true --api-version=2 --accept-multiclient exec "{{RELEASE}}/ftl" -- dev {{args}} &
   dlv_pid=$!
-  wait "$dlv_pid"
\ No newline at end of file
+  wait "$dlv_pid"
+
+# Run otel collector in a docker container to stream local (i.e. from ftl dev) signals to
+# the terminal tab where this is running. To start FTL, opepn another terminal tab and run
+# `just otel-dev` with any args you would pass to `ftl dev`. To stop the otel stream, run
+# `just otel-stop` in a third terminal tab.
+grpcPort := `cat docker-compose.yml | grep "OTLP gRPC" | sed 's/:.*//' | sed -r 's/ +- //'`
+otel-stream:
+  docker run \
+    -p {{grpcPort}}:{{grpcPort}} \
+    -p 55679:55679 \
+    otel/opentelemetry-collector:0.104.0
+
+# Stop the docker container running otel.
+otelContainerID := `docker ps -f ancestor=otel/opentelemetry-collector:0.104.0 | tail -1 | cut -d " " -f1`
+otel-stop:
+  docker stop "{{otelContainerID}}"
+
+# Run `ftl dev` with the given args after setting the necessary envar.
+otel-dev *args:
+  #!/bin/bash
+
+  gprcPort=$(cat docker-compose.yml | grep "OTLP gRPC" | sed 's/:.*//' | sed -r 's/ +- //')
```

Comment:
https://github.com/mikefarah/yq maybe easier in the future? (not an endorsement--I have not tried it!)

Reply:
Oh I see it's a comment, sorry that suggestion wouldn't work at all!

Reply:
oooo nice! I'll go ahead and merge with the current stuff so folks can use it but that would be much nicer. I'll try to get it working with the real parser in the next update

Reply:
Oh weird I just got your second comment on the comment gRPC string... yeah maybe it is actually safer to keep it like this. I'll think about it.